iterators.tex

```diff
@@ -1702,32 +1702,48 @@
-  template <@\changed{class}{_Auto}@ C> auto rbegin(C& c) -> decltype(c.rbegin());
-  template <@\changed{class}{_Auto}@ C> auto rbegin(const C& c) -> decltype(c.rbegin());
-  template <@\changed{class}{_Auto}@ C> auto rend(C& c) -> decltype(c.rend());
-  template <@\changed{class}{_Auto}@ C> auto rend(const C& c) -> decltype(c.rend());
-  template <@\changed{class}{_Auto}@ T, size_t N> reverse_iterator<T*> rbegin(T (&array)[N]);
-  template <@\changed{class}{_Auto}@ T, size_t N> reverse_iterator<T*> rend(T (&array)[N]);
-  template <@\changed{class}{_Auto}@ E> reverse_iterator<const E*> rbegin(initializer_list<E> il);
-  template <@\changed{class}{_Auto}@ E> reverse_iterator<const E*> rend(initializer_list<E> il);
-  template <@\changed{class}{_Auto}@ C> auto crbegin(const C& c) -> decltype(@\removed{std}\added{ranges\oldtxt{_v1}}@::rbegin(c));
-  template <@\changed{class}{_Auto}@ C> auto crend(const C& c) -> decltype(@\removed{std}\added{ranges\oldtxt{_v1}}@::rend(c));
+  @\oldtxt{using std::begin;}@
+  @\oldtxt{using std::end;}@
+  @\oldtxt{template <class>}@
+    @\oldtxt{concept bool _Auto = true; // \expos}@
+  @\oldoldtxt{template <class C> constexpr auto cbegin(const C\& c) noexcept(noexcept(std::begin(c)))}@
+    @\oldoldtxt{-> decltype(std::begin(c));}@
+  @\oldoldtxt{template <class C> constexpr auto cend(const C\& c) noexcept(noexcept(std::end(c)))}@
+    @\oldoldtxt{-> decltype(std::end(c));}@
+  @\oldoldtxt{template <class C> auto rbegin(C\& c) -> decltype(c.rbegin());}@
+  @\oldoldtxt{template <class C> auto rbegin(const C\& c) -> decltype(c.rbegin());}@
+  @\oldoldtxt{template <class C> auto rend(C\& c) -> decltype(c.rend());}@
+  @\oldoldtxt{template <class C> auto rend(const C\& c) -> decltype(c.rend());}@
+  @\oldoldtxt{template <class T, size_t N> reverse_iterator<T*> rbegin(T (\&array)[N]);}@
+  @\oldoldtxt{template <class T, size_t N> reverse_iterator<T*> rend(T (\&array)[N]);}@
+  @\oldoldtxt{template <class E> reverse_iterator<const E*> rbegin(initializer_list<E> il);}@
+  @\oldoldtxt{template <class E> reverse_iterator<const E*> rend(initializer_list<E> il);}@
+  @\oldoldtxt{template <class C> auto crbegin(const C\& c) -> decltype(std::rbegin(c));}@
+  @\oldoldtxt{template <class C> auto crend(const C\& c) -> decltype(std::rend(c));}@
 \end{codeblock}
 \begin{addedblock}
 \begin{codeblock}
-  template <@\oldtxt{class}\newtxt{_Auto}@ C> auto size(const C& c) -> decltype(c.size());
-  template <@\oldtxt{class}\newtxt{_Auto}@ T, size_t N> constexpr size_t size(T (&array)[N]) noexcept;
-  template <@\oldtxt{class}\newtxt{_Auto}@ E> @\newtxt{constexpr}@ size_t size(initializer_list<E> il) noexcept;
+  @\oldtxt{template <class C> auto size(const C\& c) -> decltype(c.size());}@
+  @\oldtxt{template <class T, size_t N> constexpr size_t size(T (\&array)[N]) noexcept;}@
+  @\oldtxt{template <class E> size_t size(initializer_list<E> il) noexcept;}@
+  @\newtxt{namespace \{}@
+    @\newtxt{constexpr \impdef{} begin;}@
```

Comment:
Can constexpr objects be fwd-declared like this? I don't think so. We may need an `= \impdef;` here. And do we really want "implementation defined" instead of unspecified?